src/txn/txn.c

```diff
@@ -448,6 +448,15 @@ __wt_txn_config(WT_SESSION_IMPL *session, const char *cfg[])
 		txn_global = &S2C(session)->txn_global;
 		WT_RET(__wt_txn_parse_timestamp(session, "read", &ts, &cval));
 
+		/*
+		 * Prepare transactions are supported only in timestamp build,
+		 * as it is to support mongodb exclusively.
+		 */
+		WT_RET(__wt_config_gets_def(session,
```

Comment:
I think that ignore_prepare option does not have a dependency on read_timestamp. So parsing of ignore_prepare should be independent/outside of read_timestamp parsing block.

Reply:
Prepare is supported only with timestamp builds, as prepare_timestamp is mandatory in prepare_transaction.  Hence all logic related to prepare is being kept with in timestamp macro inclusion.

Reply:
It's generally better to avoid mentioning MongoDB in cases like this - you could drop the second half of this sentence.